src/types/union.ts

```diff
@@ -965,7 +966,9 @@ export function Union(...alternatives: Rt[]): any {
   return create(
     x => {
       for (const { guard } of alternatives) if (guard(x)) return x;
-      throw new Error('No alternatives were matched');
+
+      const a = create<any>(x as never, { tag: 'union', alternatives });
+      throw validationError(`Expected a ${show(a)}, but was ${typeof x} ${JSON.stringify(x)}`);
```

Comment:
Using `JSON.stringify` is not safe because it can throw if `x` is a circular structure.

Reply:
Any suggestions for an alternative? This is especially useful for cases with API requests etc where it helps to see what the actual value was and not only the expected value.